### PR TITLE
Add branch specific namelist settings 

### PR DESCRIPTION
**Pull request by SeanBryan51**
This change allows for users to add branch specific namelist settings in the config file that get "patched" to the base namelist settings used for both branches. These additional settings are set by specifying an optional `patch` key in each realisation. Currently, this change supports the ability to add a `patch` to both realisations.

Addresses issue https://github.com/CABLE-LSM/benchcab/issues/29

### CODE REVIEW

**Comment by SeanBryan51** [comment]
@ccarouge I've linked issue #29 so that this PR will close the issue. I know a lot of the discussion was about what we do with the model output, but if there were any other requirements for the issue that I have forgotten to add please let me know

**Review by ccarouge — CHANGES_REQUESTED** [review]
One comment but with impact on a few parts of the code and the tests.

File: benchcab/task.py [hunk 1]
```diff
@@ -152,11 +164,16 @@ def setup_task(self):
...
-        self.clean_task() \
-            .fetch_files() \
-            .adjust_namelist_file()
+        self.clean_task(root_dir=root_dir) \
+            .fetch_files(root_dir=root_dir) \
+            .adjust_namelist_file(root_dir=root_dir)
+
+        if self.branch_patch:
+            # the user should not specify 'cable' in the top level of the patch dictionary
+            self.patch_namelist_file({'cable': self.branch_patch}, root_dir=root_dir)
```
[the remark this answers is not in the document]
Comment (SeanBryan51) [reply benchcab/task.py:176]: Yes I think having to include `{'cable': {......}}` in the patch is a good idea. Should we do the same for the science configurations for consistency? For example, in the [site_configs.yaml](https://github.com/CABLE-LSM/bench_example/blob/main/site_configs.yaml)
```yaml
sci0: {
    cable_user: {
        GS_SWITCH: "medlyn",
        FWSOIL_SWITCH: "Haverd2013"
    }
}
```
would become
```yaml
sci0: {
    cable: {
        cable_user: {
            GS_SWITCH: "medlyn",
            FWSOIL_SWITCH: "Haverd2013"
        }
    }
}
```

[the remark this answers is not in the document]
Comment (ccarouge) [reply benchcab/task.py:176]: Indeed we need the same in the science configurations.

[the remark this answers is not in the document]
Comment (SeanBryan51) [reply benchcab/task.py]: I have made the changes for the `patch` key: see [2006d9c](https://github.com/CABLE-LSM/benchcab/pull/43/commits/2006d9cfe261caa36cd58373d8567ea48f3bc590). I have added a separate issue for the changes for science configurations (see https://github.com/CABLE-LSM/benchcab/issues/46) as I prefer to make these changes once this PR and #44 are merged in.

**Review by ccarouge — APPROVED** [review]

**Pull request merged**